File: frontend/src/api/knowledge.ts
export interface KnowledgeBasePayload {
  id: string
  name: string
  description: string
  created_at: string
  updated_at: string
}

export interface KnowledgeBaseCreatePayload {
  name: string
  description: string
}

export interface SourceDocumentPayload {
  id: string
  knowledge_base_id: string
  name: string
  source_uri: string
  content_type: string
  status: string
  error_message: string | null
  created_at: string
}

export interface ChunkPayload {
  id: string
  knowledge_base_id: string
  source_document_id: string
  index: number
  title: string | null
  text: string
  metadata: Record<string, string>
}

export interface TextDocumentImportPayload {
  name: string
  content: string
  source_uri: string
  content_type: string
  chunk_size: number
  chunk_overlap: number
}

export interface UrlDocumentImportPayload {
  url: string
  name: string
  chunk_size: number
  chunk_overlap: number
}

export interface PdfDocumentImportPayload {
  name: string
  content_base64: string
  chunk_size: number
  chunk_overlap: number
}

export interface TextDocumentImportResult {
  document: SourceDocumentPayload
  chunks: ChunkPayload[]
  chunk_count: number
}

export interface KnowledgeIndexBuildPayload {
  knowledge_base_id: string
  document_count: number
  chunk_count: number
  status: string
  message: string
  created_at: string
}

interface ApiErrorPayload {
  message?: string
}

const API_BASE_URL = (import.meta.env.VITE_API_BASE_URL ?? 'http://localhost:8000').replace(/\/$/, '')

export async function listKnowledgeBases(): Promise<KnowledgeBasePayload[]> {
  return requestJson('/api/knowledge/bases')
}

export async function createKnowledgeBase(payload: KnowledgeBaseCreatePayload): Promise<KnowledgeBasePayload> {
  return requestJson('/api/knowledge/bases', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(payload),
  })
}

export async function deleteKnowledgeBase(knowledgeBaseId: string): Promise<void> {
  await requestEmpty(`/api/knowledge/bases/${knowledgeBaseId}`, { method: 'DELETE' })
}

export async function listKnowledgeDocuments(knowledgeBaseId: string): Promise<SourceDocumentPayload[]> {
  return requestJson(`/api/knowledge/bases/${knowledgeBaseId}/documents`)
}

export async function importTextDocument(
  knowledgeBaseId: string,
  payload: TextDocumentImportPayload,
): Promise<TextDocumentImportResult> {
  return requestJson(`/api/knowledge/bases/${knowledgeBaseId}/documents/import-text`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(payload),
  })
}

export async function importUrlDocument(
  knowledgeBaseId: string,
  payload: UrlDocumentImportPayload,
): Promise<TextDocumentImportResult> {
  return requestJson(`/api/knowledge/bases/${knowledgeBaseId}/documents/import-url`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(payload),
  })
}

export async function importPdfDocument(
  knowledgeBaseId: string,
  payload: PdfDocumentImportPayload,
): Promise<TextDocumentImportResult> {
  return requestJson(`/api/knowledge/bases/${knowledgeBaseId}/documents/import-pdf`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(payload),
  })
}

export async function listDocumentChunks(documentId: string): Promise<ChunkPayload[]> {
  return requestJson(`/api/knowledge/documents/${documentId}/chunks`)
}

export async function deleteKnowledgeDocument(documentId: string): Promise<void> {
  await requestEmpty(`/api/knowledge/documents/${documentId}`, { method: 'DELETE' })
}

export async function rebuildKnowledgeIndex(knowledgeBaseId: string): Promise<KnowledgeIndexBuildPayload> {
  return requestJson(`/api/knowledge/bases/${knowledgeBaseId}/index/rebuild`, {
    method: 'POST',
  })
}

async function requestJson<T>(path: string, init?: RequestInit): Promise<T> {
  const response = await fetch(`${API_BASE_URL}${path}`, init)
  if (!response.ok) {
    throw new Error(await readApiError(response))
  }
  return response.json() as Promise<T>
}

async function requestEmpty(path: string, init?: RequestInit): Promise<void> {
  const response = await fetch(`${API_BASE_URL}${path}`, init)
  if (!response.ok) {
    throw new Error(await readApiError(response))
  }
}

async function readApiError(response: Response): Promise<string> {
  try {
    const payload = (await response.json()) as ApiErrorPayload
    return payload.message || `请求失败：HTTP ${response.status}`
  } catch {
    return `请求失败：HTTP ${response.status}`
  }
}

File: src/ai_study_agent/api/schemas/knowledge.py
"""Knowledge API schemas."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class KnowledgeBaseCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=120)
    description: str = Field(default="", max_length=500)


class KnowledgeBaseResponse(BaseModel):
    id: str
    name: str
    description: str
    created_at: datetime
    updated_at: datetime


class TextDocumentImportRequest(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    content: str = Field(min_length=1, max_length=2_000_000)
    source_uri: str = Field(default="", max_length=500)
    content_type: str = Field(default="text/plain", max_length=80)
    chunk_size: int = Field(default=800, ge=200, le=3000)
    chunk_overlap: int = Field(default=120, ge=0, le=1000)


class UrlDocumentImportRequest(BaseModel):
    url: str = Field(min_length=8, max_length=1000)
    name: str = Field(default="", max_length=200)
    chunk_size: int = Field(default=800, ge=200, le=3000)
    chunk_overlap: int = Field(default=120, ge=0, le=1000)


class PdfDocumentImportRequest(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    content_base64: str = Field(min_length=1, max_length=15_000_000)
    chunk_size: int = Field(default=800, ge=200, le=3000)
    chunk_overlap: int = Field(default=120, ge=0, le=1000)


class SourceDocumentResponse(BaseModel):
    id: str
    knowledge_base_id: str
    name: str
    source_uri: str
    content_type: str
    status: str
    error_message: str | None
    created_at: datetime


class ChunkResponse(BaseModel):
    id: str
    knowledge_base_id: str
    source_document_id: str
    index: int
    title: str | None
    text: str
    metadata: dict[str, str]


class TextDocumentImportResponse(BaseModel):
    document: SourceDocumentResponse
    chunks: list[ChunkResponse]
    chunk_count: int


class KnowledgeIndexBuildResponse(BaseModel):
    knowledge_base_id: str
    document_count: int
    chunk_count: int
    status: str
    message: str
    created_at: datetime
